src/components/layers/displayLayers.jsx

```diff
@@ -0,0 +1,81 @@
+import React, { useEffect, useRef, useState } from 'react'
+import { useMap } from '../../contexts/mapContext/mapContext';
+import { getLayers } from './getLayers';
+import VectorLayer from 'ol/layer/Vector';
+import VectorSource from 'ol/source/Vector';
+import Style from 'ol/style/Style';
+import Icon from 'ol/style/Icon';
+import { Box } from '@mui/material';
+import PopoverComponent from '../popover/PopoverComponent';
+import GeoJSON from 'ol/format/GeoJSON.js';
+import { Overlay } from 'ol';
+import { getArrayOfAllLayers, getArrayOfVectorLayersWithoutDrawing } from './getArrayOfLayers';
+
+const DisplayLayers = () => {
+    const mapRef = useRef(useMap());

```

Comment:
You can simply use either `const map = useMap() or const { map } = useMap() `depending on how the hook returns the value.